### PR TITLE
update Baseclient.ts documentation

### PR DESCRIPTION
**Pull request by Elli610**
related to #152 

The "account" argument in "compactBytesForOperation" is unused in the function but we cannot remove it because the npm run build won't work: other functions call "compactBytesForOperation". 
This issue needs to be fixed in another pr. Related issue : #154

### CODE REVIEW

File: src/web3/BaseClient.ts [hunk 1]
```diff
@@ -69,26 +78,54 @@ export class BaseClient {
...
+   * @remarks
+   * This method chooses the provider to use for a given rpc method.
+   *  - If the rpc method is about getting or sending data to the blockchain,
+   *    it will choose a public provider.
+   *  - If the rpc method is meant to be used by the node itself, it will choose a private provider.
+   *  - An error is thrown if no provider is found for the rpc method
+   *
+   * @param requestMethod - The rpc method to find the provider for
+   *
+   *  @returns The provider for the rpc method
```
Comment (0xSwapFeeder) [inline src/web3/BaseClient.ts:127]:
```suggestion
   * @returns The provider for the rpc method
```

File: src/web3/BaseClient.ts [hunk 2]
```diff
@@ -126,6 +163,18 @@ export class BaseClient {
...
+   * Converts a json rpc call to a promise that resolves as a JsonRpcResponseData
+   *
+   * @privateRemarks
+   * If there is an error while sending the request, the function catches the error, the isError
+   * property is set to true, the result property set to null, and the error property set to a
+   * new Error object with a message indicating that there was an error.
+   *
+   * @param resource - The rpc method to call
+   * @param params - The parameters to pass to the rpc method
+   * @returns A promise that resolves as a JsonRpcResponseData
```
Comment (0xSwapFeeder) [inline src/web3/BaseClient.ts:176]:
```suggestion
   *
   * @returns A promise that resolves as a JsonRpcResponseData
```

File: src/web3/BaseClient.ts [hunk 3]
```diff
@@ -186,7 +244,15 @@ export class BaseClient {
...
 
-  /** compact bytes payload per operation */
+  /**
+   * Compacts bytes payload per operation
+   *
+   * @param data - The operation data
+   * @param opTypeId - The operation type id
+   * @param account - The account used
+   * @param expirePeriod - The expire period
+   * @return The compacted bytes payload
```
Comment (0xSwapFeeder) [inline src/web3/BaseClient.ts:254]:
```suggestion
   *
   * @return The compacted bytes payload
```

**Review by 0xSwapFeeder — CHANGES_REQUESTED** [review]

**Review by 0xSwapFeeder — APPROVED** [review]

**Review by Ben-Rey — APPROVED** [review]

**Comment by Ben-Rey** [comment]
@Elli610 can you check why the CI Fail ?

**Pull request merged**